tools/run-all-tests.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/env bash
+
+set -euo pipefail
+
+script_dir="$(cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")" && pwd -P)/"
+stdb_root="$(realpath "$script_dir/../")"
+
+# Find cargo.
+cargo=$(which cargo)
+if [ ! -x "$cargo" ] ; then
+    echo "Could not find 'cargo' executable, please install it."
+    exit 1
+fi
+
+# Find dotnet.
+dotnet=$(which dotnet)
+if [ -x "$dotnet" ] ; then
+    dotnet_version=$($dotnet --version 2>&1 | awk -F '.' '{print $1}')
+    if [ "$dotnet_version" -lt 8 ] ; then
+        dotnet=""
+        echo "Could not find 'dotnet' version 8 executable ($dotnet is version $($dotnet --version 2>&1)), skipping dotnet tests."
+    fi
+fi
+
+# Find python3.
+python=$(which python3)
+if [ ! -x "$python" ] ; then
+    python=$(which python)
+    if [ -x "$python" ] ; then
+        python="$python"
+    fi
+fi
+if [ -x "$python" ] ; then
+    py_version=$($python --version 2>&1 | awk '{print $2}' | awk -F '.' '{print $1}')
+    if [ "$py_version" -lt 3 ] ; then
+        python=""
+        echo "Could not find python3 executable (tried: python and python3), skipping smoketests."
+    fi
+fi
+
+set -euox pipefail
+
+cd "$stdb_root"
+tools/clippy.sh
+cargo test --all
+if [ -x "$python" ] ; then
+    $python -m smoketests
+fi
+if [ -x "$dotnet" ] ; then
+    $dotnet test crates/bindings-csharp
+fi
```

Comment:
won't the tests themselves generate the errors about `dotnet` not being found, being the wrong version, etc?

if `dotnet` is in the path, then this would work.. couldn't we just.. run `dotnet test crates/bindings-csharp` without really bothering to do any looking for it in the first place?